File: 114-二叉树展开为链表.cpp
class Solution {
public:
    void flatten(TreeNode* root) {
        if(root){
            stack<TreeNode*> mystack;
            mystack.push(root);
            TreeNode* cur_node;

            while(!mystack.empty()){
                cur_node = mystack.top();
                mystack.pop();
                if(cur_node->right)
                    mystack.push(cur_node->right);
                if(cur_node->left)
                    mystack.push(cur_node->left);
                cur_node->left = NULL;
                cur_node->right = mystack.empty() ? NULL : mystack.top();
            }
        }
    }
};

File: 114-二叉树展开为链表.py
class Solution:
    def flatten(self, root: TreeNode) -> None:
        """
        Do not return anything, modify root in-place instead.
        """
        if root:
            mystack = [root]

            while mystack:
                node = mystack.pop()
                if node.right:
                    mystack.append(node.right)
                if node.left:
                    mystack.append(node.left)
                node.right = mystack[-1] if mystack else None
                node.left = None
